src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -105,6 +119,16 @@ public function toInputArgument(): InputArgument
      */
     public function resolveValue(InputInterface $input): mixed
     {
-        return $input->getArgument($this->name);
+        $value = $input->getArgument($this->name);
+
+        if (is_subclass_of($this->typeName, \BackedEnum::class) && (is_string($value) || is_int($value))) {
+            try {
+                $value = ($this->typeName)::from($value);
+            } catch (\ValueError) {
+                throw InvalidArgumentException::fromEnumValue($this->name, $value, $this->suggestedValues);
+            }
+        }
+
+        return $value;
```

Comment:
```suggestion
        if (!is_subclass_of($this->typeName, \BackedEnum::class) || (!is_string($value) && !is_int($value))) {
            return $value
        }

        return ($this->typeName)::tryFrom($value) ?? throw InvalidArgumentException::fromEnumValue($this->name, $value, $this->suggestedValues);
```

Reply:
Using early return is a good idea. I prefers the 2nd suggestion that handle the backed enum in the if (as a special case).
```suggestion
        if (is_subclass_of($this->typeName, \BackedEnum::class) && (is_string($value) || is_int($value))) {
            return ($this->typeName)::tryFrom($value) ?? throw InvalidArgumentException::fromEnumValue($this->name, $value, $this->suggestedValues);
        }

        return $value;
```